File: test_data/code/java/1716-calculate-money-in-leetcode-bank.java
class Solution {
    public int totalMoney(int n) {
        int weeks = n / 7;
        int remainDays = n % 7;
        int total = 0;
        
        for (int i = 1; i <= weeks; i++) {
            total += 28 + i; // 28 = 7 * 4 (1 + 2 + ... + 7)
        }
        
        for (int i = 1; i <= remainDays; i++) {
            total += i + 7; // on the first monday after weeks, he will start from 7
        }
        
        return total;
    }
}

File: test_data/code/python/1716-calculate-money-in-leetcode-bank.py
Python
class Solution:
    def totalMoney(self, n: int) -> int:
        n -= 1     # subtract 1 because Monday is 0
        week = n // 7
        remainder = n % 7
        return week * 28 + sum(range(1, 8)) + sum(range(1, remainder + 1))